src/qos_core/src/protocol/processor.rs

```diff
@@ -52,8 +52,9 @@ impl ProtocolProcessor {
 		self.state.read().await.get_phase()
 	}
 
-	/// Sets the client timeout value for the `app_client`
+	/// Sets the client timeout value for the `app_client`, maximum allowed value is `u16::MAX` milliseconds
 	pub fn set_client_timeout(&mut self, timeout: Duration) {
+		assert!(timeout.as_millis() < u16::MAX.into(), "client timeout > 65s");
```

Comment:
I actually think this will likely be too restrictive (e.g. zk proof gen). But we can cross that bridge when we get there